sql/plsql/PlSqlParser.g4

```diff
@@ -4392,7 +4392,21 @@ table_alias
     ;
 
 where_clause
-    : WHERE (CURRENT OF cursor_name | expression)
+    : WHERE ((CURRENT OF cursor_name | expression) | all_some_any_where_statment)
+    ;
+
+all_some_any_where_statment
```

Comment:
Either use the abbreviation `stmt` or write it as `statement`. `statment` is somewhere in between.